cmake/Modules/FindComputeCpp.cmake

```diff
@@ -401,19 +391,51 @@ function(add_sycl_to_target)
     "${multi_value_args}"
     ${ARGN}
   )
-  set(fileCounter 0)
-  # Add custom target to run compute++ and generate the integration header
-  foreach(sourceFile ${SDK_ADD_SYCL_SOURCES})
-    if(NOT IS_ABSOLUTE ${sourceFile})
-      set(sourceFile "${CMAKE_CURRENT_SOURCE_DIR}/${sourceFile}")
+
+  # If the CXX compiler is set to compute++ enable the driver.
+  get_filename_component(cmakeCxxCompilerFileName "${CMAKE_CXX_COMPILER}" NAME)
+  if("${cmakeCxxCompilerFileName}" STREQUAL "compute++")
+    if(MSVC)
+      message(FATAL_ERROR "The compiler driver is not supported by this system,
+                           revert the CXX compiler to your default host compiler.")
     endif()
-    __build_ir(
-      TARGET     ${SDK_ADD_SYCL_TARGET}
-      SOURCE     ${sourceFile}
-      COUNTER    ${fileCounter}
-    )
-    MATH(EXPR fileCounter "${fileCounter} + 1")
-  endforeach()
+
+    get_target_property(includeAfter ${SDK_ADD_SYCL_TARGET} COMPUTECPP_INCLUDE_AFTER)
+    if(includeAfter)
+      list(APPEND COMPUTECPP_USER_FLAGS -fsycl-ih-last)
+    endif()
+    list(INSERT COMPUTECPP_DEVICE_COMPILER_FLAGS 0 -sycl-driver)
+    # Prepend COMPUTECPP_DEVICE_COMPILER_FLAGS and append COMPUTECPP_USER_FLAGS
+    foreach(prop COMPILE_OPTIONS INTERFACE_COMPILE_OPTIONS)
```

Comment:
This whole foreach loop is supposed to be equivalent to:
```
target_compile_options(${SDK_ADD_SYCL_TARGET} BEFORE PUBLIC ${COMPUTECPP_DEVICE_COMPILER_FLAGS})
target_compile_options(${SDK_ADD_SYCL_TARGET} PUBLIC ${COMPUTECPP_USER_FLAGS})
```
I don't understand why the `BEFORE` option is not prepending the flags as I think it should. If we don't prepend the default `-O2` could override the user's optimization level.
Should we only add `-O2` for the 2 steps compilation instead?

Reply:
I don't really mind. `O2` is definitely needed, right enough. Can't you `list(PREPEND COMPUTECPP_USER_FLAGS -O2 etc)`?

Reply:
It is an issue if you want to debug and give `O0`.
`list(PREPEND ...)` won't make it simpler, I would still need to do the loop for both `COMPILE_OPTIONS` and `INTERFACE_COMPILE_OPTIONS`, to read the property and to set it again.